Refactor networking

pacjac opened the pull request:
Mostly directory structure and convenience refactoring.

1. Implemented Publisher and Subscriber abstract base class for the sake of type hints.
2. Moved mqtt, zmq and serial wrappers to msb/networking to not clutter msb/ too much
3. Added get_publisher and get_subscriber factory functions. Use with 

```from msb.networking import get_publisher, get_subscriber
zmq_pub = get_publisher("zmq")
mqtt_sub = get_subscriber("mqtt", "/sensor/temp")```

4. msb/mqtt and msb/zmq have NOT been deleted but are considered deprecated.
5. Added serial pub/sub draft
6. Added msb/network/packer which provides a common interface for serializing and deserializing. Not fully tested.
7. Started work on unittests with Mock objects.

k323r commented on msb/network/pubsub/factories.py:
```diff
@@ -0,0 +1,63 @@
+import os
+
+from .types import Publisher, Subscriber
```
nice pattern, had to read up on it, learned a ton :+1:

k323r reviewed (commented):
Impressive rewrite! No changes requested to the core code from my side, however, before merging, I'd like to make sure that the legacy publisher/subscriber model as currently used in our services still works.

kawindme commented on msb/network/serial/publisher.py:
```diff
@@ -0,0 +1,43 @@
+            port=self.config.port,
+            baudrate=self.config.baudrate,
+            bytesize=self.config.bytesize,
+            parity=serial.PARITY_NONE,
```
Parity is defined in SerialConf but not used here.

pacjac replied:
Removed from config.

pacjac commented:
Addressed comments. Added functionality to import types from msb.network

e.g.

```
from msb.network import Subscriber
from msb.network import ZMQ_Publisher
from msb.network import MQTT_Subscriber
```
